Symfony/CS/Fixer/Contrib/ModernizeTypesCastingFixer.php

```diff
@@ -0,0 +1,150 @@
+                    if (!$tokens[$paramContentIndex]->isGivenKind(T_WHITESPACE)) {
+                        ++$countParamTokens;
+                    }
+
+                    // skip (...) constructs
+                    if ('(' === $tokens[$paramContentIndex]->getContent()) {
+                        $paramContentIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $paramContentIndex);
+                        continue;
+                    }
+                    // check if commas are available
+                    if (',' === $tokens[$paramContentIndex]->getContent()) {
+                        $hasCommas = true;
+                    }
+                }
+                $preserveParenthesises = $countParamTokens > 1;
+
+                // special case: intval with 2 parameters shall not be processed
+                if ($hasCommas && 'intval' === $functionIdentity) {
+                    continue;
+                }
+
+                // handle function reference with namespaces
+                if ($prevToken->isGivenKind(T_NS_SEPARATOR)) {
+                    $twicePrevTokenIndex = $tokens->getPrevMeaningfulToken($prevTokenIndex);
+                    $twicePrevToken = $tokens[$twicePrevTokenIndex];
+                    if ($twicePrevToken->isGivenKind(array(T_NEW, T_STRING, CT_NAMESPACE_OPERATOR))) {
+                        continue;
+                    }
+
+                    // get rid of root namespace when it used
+                    $tokens->removeTrailingWhitespace($prevTokenIndex);
+                    $tokens[$prevTokenIndex]->clear();
+                }
+
+                // perform transformation
+                $replacement = array(
+                    new Token($castingTokens[$functionIdentity]),
+                    new Token(array(T_WHITESPACE, ' ')),
+                );
+                if (!$preserveParenthesises) {
+                    // closing parenthesis removed with leading spaces
+                    $tokens->removeLeadingWhitespace($closeParenthesis);
+                    $tokens[$closeParenthesis]->clear();
+
+                    // opening parenthesis removed with trailing spaces
+                    $tokens->removeLeadingWhitespace($openParenthesis);
+                    $tokens->removeTrailingWhitespace($openParenthesis);
+                    $tokens[$matches[1]]->clear();
+                } else {
+                    // we'll need to provide a space after a casting operator
+                    $tokens->removeTrailingWhitespace($functionName);
+                }
+                $tokens->overrideRange($functionName, $functionName, $replacement);
+
+                // nested transformations support
+                $currIndex = $functionName;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Replaces intval, floatval, strval, boolval functions calls with according type casting operator.';
```

Comment:
77d08cf : takes in account if function is defined in analysis scope.

Is the location for new utility class fine?